### PR TITLE
[SYCL] Deny pragma spelling for SYCL-specific attributes

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by MrSidims** [comment]
TBH, I think we can also remove 'pragma spelling' for `IntelReqdSubGroupSize` as well. Yet I would prefer to leave `ReqdWorkGroupSize` attribute untouched. @erichkeane what do you think?

**Comment by erichkeane** [comment]
> 
> 
> TBH, I think we can also remove 'pragma spelling' for `IntelReqdSubGroupSize` as well. Yet I would prefer to leave `ReqdWorkGroupSize` attribute untouched. @erichkeane what do you think?

I don't have a strong opinion, but I don't know how heavy the pragma spelling for thoseis used for OpenCL.  Do we know?

**Comment by MrSidims** [comment]
> > TBH, I think we can also remove 'pragma spelling' for `IntelReqdSubGroupSize` as well. Yet I would prefer to leave `ReqdWorkGroupSize` attribute untouched. @erichkeane what do you think?
> 
> I don't have a strong opinion, but I don't know how heavy the pragma spelling for thoseis used for OpenCL. Do we know?

Except of loop pragmas (unroll, ivdep etc), I have never encountered any pragmas in OpenCL.

**Comment by erichkeane** [comment]
> 
> 
> > > TBH, I think we can also remove 'pragma spelling' for `IntelReqdSubGroupSize` as well. Yet I would prefer to leave `ReqdWorkGroupSize` attribute untouched. @erichkeane what do you think?
> > 
> > 
> > I don't have a strong opinion, but I don't know how heavy the pragma spelling for thoseis used for OpenCL. Do we know?
> 
> Except of loop pragmas (unroll, ivdep etc), I have never encountered any pragmas in OpenCL.

Then I think we can remove the pragma spelling for IntelReqdGroupSize, and if there's a regression revert that part.

Whats the logic for wanting to leave `ReqWorkGroupSize` untouched?

**Comment by MrSidims** [comment]
> Whats the logic for wanting to leave `ReqWorkGroupSize` untouched?

It's not Intel-specific attribute coming from OpenCL core spec itself.
At the same time quick googling showed me this: 

> No results found for "pragma reqd_work_group_size"

So it might be safe to remove the spelling for it as well.

**Comment by MrSidims** [comment]
> > Whats the logic for wanting to leave `ReqWorkGroupSize` untouched?
> 
> It's not Intel-specific attribute coming from OpenCL core spec itself.
> At the same time quick googling showed me this:
> 
> > No results found for "pragma reqd_work_group_size"
> 
> So it might be safe to remove the spelling for it as well.

@keryell may be you know if any of your customers use 'pragma' spelling?

File: clang/test/Misc/pragma-attribute-supported-attributes-list.test [hunk 1]
```diff
@@ -136,14 +135,10 @@
 // CHECK-NEXT: Pointer (SubjectMatchRule_record_not_is_union)
 // CHECK-NEXT: ReleaseHandle (SubjectMatchRule_variable_is_parameter)
 // CHECK-NEXT: RenderScriptKernel (SubjectMatchRule_function)
-// CHECK-NEXT: ReqdWorkGroupSize (SubjectMatchRule_function)
```
Comment (Fznamznon) [inline clang/test/Misc/pragma-attribute-supported-attributes-list.test:139]: I don't have OpenCL background, so I cannot say is correct change or not, but this particular attribute `ReqdWorkGroupSize` is OpenCL specific and it is added not by us.
Suggest you go through community with change for this attribute first.
I think you will get answers on your questions regarding this attribute.

**Comment by keryell** [comment]
> @keryell may be you know if any of your customers use 'pragma' spelling?

Traditional Xilinx tools use a lot of `#pragma` but not the SYCL path.

While I am an advocate of no user-visible `#pragma` or attribute in SYCL to use only C++ decorative functions (think Python @decorator...) or classes instead, `#pragma` play even less nicely with C++ than attributes. :-( So at the end I do not have issues with having only attributes, as a less evil solution. :-)

File: clang/test/Misc/pragma-attribute-supported-attributes-list.test (hunk 1 above)
Comment (MrSidims) [reply clang/test/Misc/pragma-attribute-supported-attributes-list.test]: Ok, lets stick with leaving reqd_work_group_size attribute alone.

**Pull request merged**